Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
```

Comment:
We're going to combine it with a domain name and ultimately call `new MailAddress(...)` where `MailAddress` is the type `System.Net.Mail.MailAddress`.  And actually looking into that class, I see it will throw a `FormatException` if the provided email address is not in the proper format, which seems to include non-US ASCII characters.  Digging into that a bit, `MailAddressParser` actually handles the validation logic for `MailAddress`, unfortunately it's internal so can't be used as-is.  But seeing that we're planning to use `EmailAddressLocalPart` to construct `MailAddress` and given that the latter will validate the email address, I think we are obliged to have `EmailAddressLocalPart` enforce the same rules if it's to be useful.